ee/server/service/maintained_apps.go

```diff
@@ -62,51 +63,41 @@ func (svc *Service) AddFleetMaintainedApp(
 		return 0, ctxerr.Wrap(ctx, err, "getting maintained app by id")
 	}
 
+	app, err = maintained_apps.Hydrate(ctx, app)
+	if err != nil {
+		return 0, ctxerr.Wrap(ctx, err, "hydrating app from manifest")
+	}
+
 	// Download installer from the URL
-	timeout := maintainedapps.InstallerTimeout
+	timeout := maintained_apps.InstallerTimeout
 	if v := os.Getenv("FLEET_DEV_MAINTAINED_APPS_INSTALLER_TIMEOUT"); v != "" {
 		timeout, _ = time.ParseDuration(v)
 	}
 
 	client := fleethttp.NewClient(fleethttp.WithTimeout(timeout))
-	installerTFR, filename, err := maintainedapps.DownloadInstaller(ctx, app.InstallerURL, client)
+	installerTFR, filename, err := maintained_apps.DownloadInstaller(ctx, app.InstallerURL, client)
 	if err != nil {
 		return 0, ctxerr.Wrap(ctx, err, "downloading app installer")
 	}
 	defer installerTFR.Close()
 
-	extension, err := maintainedapps.ExtensionForBundleIdentifier(app.BundleIdentifier)
-	if err != nil {
-		return 0, ctxerr.Errorf(ctx, "getting extension from bundle identifier %q", app.BundleIdentifier)
-	}
+	h := sha256.New()
+	_, _ = io.Copy(h, installerTFR) // writes to a Hash can never fail
+	gotHash := hex.EncodeToString(h.Sum(nil))
 
-	// Validate the bytes we got are what we expected, if homebrew supports
-	// it, the string "no_check" is a special token used to signal that the
-	// hash shouldn't be checked.
+	// Validate the bytes we got are what we expected, if a valid SHA is supplied
 	if app.SHA256 != noCheckHash {
-		h := sha256.New()
-		_, _ = io.Copy(h, installerTFR) // writes to a Hash can never fail
-		gotHash := hex.EncodeToString(h.Sum(nil))
-
 		if gotHash != app.SHA256 {
 			return 0, ctxerr.New(ctx, "mismatch in maintained app SHA256 hash")
 		}
-
-		if err := installerTFR.Rewind(); err != nil {
-			return 0, ctxerr.Wrap(ctx, err, "rewind installer reader")
-		}
-	}
-
-	// Fall back to the filename if we weren't able to extract a filename from the installer response
-	if filename == "" {
-		filename = app.Name
+	} else { // otherwise set the app hash to what we downloaded so storage writes correctly
+		app.SHA256 = gotHash
 	}
 
-	// The UI requires all filenames to have extensions. If we couldn't get
-	// one, use the extension we extracted prior
-	if filepath.Ext(filename) == "" {
-		filename = filename + "." + extension
+	if err := installerTFR.Rewind(); err != nil {
+		return 0, ctxerr.Wrap(ctx, err, "rewind installer reader")
 	}
+	extension := filepath.Ext(filename)[1:]
```

Comment:
nit: could we do `strings.TrimLeft(filepath.Ext(filename), ".")` to be extra sure we won't panic here?

Reply:
Works for me. Fix incoming.